src/controllers/wallet/atomic-swap/tx-proposal.controller.js

```diff
@@ -156,6 +157,33 @@ async function buildTxProposal(req, res) {
   }
 }
 
+/**
+ * Fetches all proposal data from the Atomic Swap Service for a specific identifier
+ */
+async function fetchFromService(req, res) {
+  if (!constants.SWAP_SERVICE_FEATURE_TOGGLE) {
+    res.status(405).json({ success: false, error: 'Method not allowed' });
+    return;
+  }
+  const { password, proposal_id: proposalId } = req.body;
+  // A service mediated request must have the identifier and the password as parameters
+  if (!password || !proposalId) {
+    res.status(400).json({
+      success: false,
+      error: 'Parameters proposal_id and password are mandatory when requesting from service'
+    });
+    return;
+  }
```

Comment:
Don't we validate that with the schema in the route?

Reply:
Yes, we do! Removed the redundant validation on [a8d69f0](https://github.com/HathorNetwork/hathor-wallet-headless/pull/277/commits/a8d69f00ae77210cdc91ae491d82dcaf861a9e8a)